### PR TITLE
Fixes #255

### LINKED ISSUE #255: Executer, Variables in string are only replaced in SAY
When you put a variable in ask like "How old are you <gd.name> ?"

The variable is not replaced. It is only works for SAY.

### PR DESCRIPTION
**Pull request by Jeremi360**
Extracting the do_variables () function and applying it to all making sense strings in Executer.gd to fix #255

### CODE REVIEW

File: Test/TestExecuter/TestAsk/TestAsk.tscn [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [gd_scene load_steps=2 format=3 uid="uid://c65gxwwshluof"]
 
-[ext_resource type="Script" path="res://Test/TestParser/TestAsk/TestAsk.gd" id="1"]
+[ext_resource type="Script" path="res://Test/TestExecuter/TestAsk/TestAsk.gd" id="1_q58ae"]
```
Comment (theludovyc) [inline Test/TestExecuter/TestAsk/TestAsk.tscn:3]: Thank you :) !

File: Test/TestParser/TestSay/TestSay.gd [hunk 2]
```diff
@@ -51,27 +51,7 @@ func test_say_character():
...
-	assert_false(parsed_array.is_empty())
-	
-	assert_true(parsed_script.get("labels", [""]).is_empty())
-
-	assert_true(parsed_array[0][0] == "SAY")
-
-	var result = parsed_array[0][1]
-	
-	assert_eq(result["text"], "My name is <sy.name>")
-
```
Comment (theludovyc) [inline Test/TestParser/TestSay/TestSay.gd]: Why do you remove it ? It is permit to certified the parser recognize string with variable and do not broke them

File: addons/Rakugo/.gitattributes
Comment (theludovyc) [inline addons/Rakugo/.gitattributes:1]: Add it in a clean PR please :)

File: addons/Rakugo/.gitignore
Comment (theludovyc) [inline addons/Rakugo/.gitignore:1]: Add it in a clean PR please :)

File: addons/Rakugo/icons/16x16Icon.png.import
Comment (theludovyc) [inline addons/Rakugo/icons/16x16Icon.png.import:1]: What is that ?

File: addons/Rakugo/plugin.gd
Comment (theludovyc) [inline addons/Rakugo/plugin.gd:1]: Do it in another PR please :)

File: project.godot
Comment (theludovyc) [inline project.godot:1]: I do not understand the updates

File: addons/Rakugo/Rakugo.gd [hunk 3]
```diff
@@ -86,6 +86,9 @@ func get_variable(var_name: String):
 			if store_manager.variables.has(var_name):
 				variable = store_manager.variables.get(var_name)
 			mutex.unlock()
+			
+			if variable is String:
+				return executer.replace_variables(variable)
```
Comment (theludovyc) [inline addons/Rakugo/Rakugo.gd]: Please create an issue for that, we need to talk about that, I do not think it is a great idea. It sounds like but I am not conviced

File: addons/Rakugo/Rakugo.gd [hunk 4]
```diff
@@ -210,6 +213,10 @@ func get_character_variable(character_tag: String, var_name: String):
 		push_error("Available variables are: " + str(character.keys()))
 
 	mutex.unlock()
+
+	if character_variable is String:
+		return executer.replace_variables(character_variable)
+
```
Comment (theludovyc) [inline addons/Rakugo/Rakugo.gd]: In the same new issue please :)

File: Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd
Comment (theludovyc) [inline Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd:1]: Please put it in res://Test/TestParser/TestVariables/TestVariables.gd

**Review by theludovyc — CHANGES_REQUESTED** [review]

File: Test/TestParser/TestSay/TestSay.gd (hunk 2 above)
Comment (Jeremi360) [reply Test/TestParser/TestSay/TestSay.gd]: No, it dosen't do this as Parser doesn't handel variables at strings at all, Executer dose that. So it can react to change of variable values in run time.

File: addons/Rakugo/.gitattributes
Comment (Jeremi360) [reply addons/Rakugo/.gitattributes:1]: Okey, I will move that to separted PR

File: addons/Rakugo/.gitignore
Comment (Jeremi360) [reply addons/Rakugo/.gitignore:1]: Okey, I will move that to separted PR

File: addons/Rakugo/icons/16x16Icon.png.import
Comment (Jeremi360) [reply addons/Rakugo/icons/16x16Icon.png.import:1]: For some reason Godot remiported one icon.
I think that this is some old icon that was used in past for some node class as in some godot 3.x svg icons in when used in class names wasn't scalled down.
Is no longer need so I will remove it separeted PR with .git files in inside addons/Rakugo dir.

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (Jeremi360) [reply addons/Rakugo/Rakugo.gd]: This is the same issue it is need so this can work dynamic as it should:
`a = "my name is <player.name>"`

File: addons/Rakugo/Rakugo.gd (hunk 4 above)
Comment (Jeremi360) [reply addons/Rakugo/Rakugo.gd]: The same as above, but if your realy want, I can make this a separted issue

File: addons/Rakugo/plugin.gd
Comment (Jeremi360) [reply addons/Rakugo/plugin.gd:1]: Okey

File: project.godot
Comment (Jeremi360) [reply project.godot:1]: Lines 35 - 37 changes are, because someone previously opened the project in the Godot C # version and not GDScript only.

Lines 43 - 47 I was the last to try to import blender files into another project, which failed. So I turned off those setting, but Godot adds it to every project as we see.
I will remove it.

File: Test/TestParser/TestSay/TestSay.gd (hunk 2 above)
Comment (theludovyc) [reply Test/TestParser/TestSay/TestSay.gd]: Yeah, it does not handle the replace of the variable, but parser parse. And we need to test it, to be sure it work as well intented and produce a result that can be used by the executer.

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (theludovyc) [reply addons/Rakugo/Rakugo.gd]: Maybe you think it is needed in this case
```
test_var = "a"
test_var_b = "<test_var>b"
```
But remember it is a string not a value like int or float, so before assign it, we replace the variables. You do it perfectly well by create the func replace_variables(), and use it on each string. So you do not need to update the get_variable() func.

**Review by theludovyc — CHANGES_REQUESTED** [review]

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (Jeremi360) [reply addons/Rakugo/Rakugo.gd]: Okey I will test if this work with out updates to `get_variable()`

File: Test/TestParser/TestSay/TestSay.gd (hunk 2 above)
Comment (Jeremi360) [reply Test/TestParser/TestSay/TestSay.gd]: Okey, I will restore it.

File: Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd
Comment (Jeremi360) [reply Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd:1]: But this particular test checks if the variable has been changed - Exectuer does that, not Parser

File: project.godot
Comment (Jeremi360) [reply project.godot:1]: Done 😁

File: Test/TestParser/TestSay/TestSay.gd (hunk 2 above)
Comment (Jeremi360) [reply Test/TestParser/TestSay/TestSay.gd]: Restored :grin:

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (Jeremi360) [reply addons/Rakugo/Rakugo.gd]: > But remember it is a string not a value like int or float, so before assign it, we replace the variables. You do it perfectly well by create the func replace_variables(), and use it on each string. So you do not need to update the get_variable() func.

But what with loading save ? then we need to use replace_variables() I think @theludovyc

File: addons/Rakugo/.gitattributes
Comment (Jeremi360) [reply addons/Rakugo/.gitattributes:1]: Retored I will soon make sparted PR for it and other files

File: addons/Rakugo/.gitignore
Comment (Jeremi360) [reply addons/Rakugo/.gitignore:1]: Retored I will soon make sparted PR for it and other files

File: addons/Rakugo/icons/16x16Icon.png.import
Comment (Jeremi360) [reply addons/Rakugo/icons/16x16Icon.png.import:1]: Revertd I will soon make sparted PR for removing it and orignal png it and other files

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (Jeremi360) [reply addons/Rakugo/Rakugo.gd]: image

File: Test/TestParser/TestSay/TestSay.gd (hunk 2 above)
Comment (theludovyc) [reply Test/TestParser/TestSay/TestSay.gd]: Outdated, I put it to reloved

File: addons/Rakugo/.gitattributes
Comment (theludovyc) [reply addons/Rakugo/.gitattributes:1]: Outdated, I put it to reloved

File: addons/Rakugo/.gitignore
Comment (theludovyc) [reply addons/Rakugo/.gitignore:1]: Outdated, I put it to reloved

File: addons/Rakugo/icons/16x16Icon.png.import
Comment (theludovyc) [reply addons/Rakugo/icons/16x16Icon.png.import:1]: Outdated, I put it to reloved

File: addons/Rakugo/plugin.gd
Comment (theludovyc) [reply addons/Rakugo/plugin.gd:1]: Outdated, I put it to reloved

File: Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd
Comment (theludovyc) [reply Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd:1]: Outdated, I put it to reloved

Comment (theludovyc) [inline Test/TestExecuter/TestDynamicVars/TestDynamicVars.gd:1]: You can put it in Test/TestParser/TestVariables instead of recreate a test

If you want you can move TestVariables in TestExecuter

File: Test/TestExecuter/TestMenu/TestMenu.gd
Comment (theludovyc) [inline Test/TestExecuter/TestMenu/TestMenu.gd:1]: Changes in this file are really strange, you should rebase :)

File: addons/Rakugo/Rakugo.gd (hunk 3 above)
Comment (theludovyc) [inline addons/Rakugo/Rakugo.gd]: If replace_variables() is used in Rakugo.gd, move it from Executer.gd. It is better than call if from executer.
Plus user can be interested by using it.

**Review by theludovyc — CHANGES_REQUESTED** [review]

**Comment by Jeremi360** [comment]
This was auto-closed by mistake

File: Test/TestParser/TestVariables/TestVariables.gd
Comment (theludovyc) [inline Test/TestParser/TestVariables/TestVariables.gd:1]: You need to restore this file from main

File: addons/Rakugo/lib/systems/Executer.gd [hunk 5]
```diff
@@ -269,47 +273,8 @@ func do_execute_script(parameters:Dictionary):
 					else:
 						value = float(value)
 
-				var assignment = result["assignment"]
-				
-				var lvar_name = result["lvar_name"]
+				Rakugo.set_variable(result["lvar_name"], value)
```
Comment (theludovyc) [inline addons/Rakugo/lib/systems/Executer.gd]: You gonna introduce regression here. Maybe copy/paste the code from main is the easiest way :)

File: addons/Rakugo/lib/systems/Parser.gd
Comment (theludovyc) [inline addons/Rakugo/lib/systems/Parser.gd:1]: You need to restore this file from main

File: Test/TestParser/TestVariables/TestVariables.rk
Comment (theludovyc) [inline Test/TestParser/TestVariables/TestVariables.rk:1]: You need to restore this file from main

**Review by theludovyc — CHANGES_REQUESTED** [review]

File: Test/TestExecuter/TestMenu/TestMenu.gd
Comment (theludovyc) [reply Test/TestExecuter/TestMenu/TestMenu.gd:1]: Resolved

File: .gut_editor_config.json
Comment (theludovyc) [inline .gut_editor_config.json:1]: Do not push this file please :)

File: addons/Rakugo/lib/systems/Executer.bak.gd
Comment (theludovyc) [inline addons/Rakugo/lib/systems/Executer.bak.gd:1]: Do not push this file please :)

**Review by theludovyc — CHANGES_REQUESTED** [review]

**Review by theludovyc — APPROVED** [review]
Yeah !!! 🥳
